### PR TITLE
fix failing spec tests

### PR DESCRIPTION
**Pull request by Phil-Friderici**
fix failing spec tests

### CODE REVIEW

**Comment by Phil-Friderici** [comment]
finally it works :)

File: .fixtures.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ fixtures:
       ref: '4.6.0'
     common:
       repo: 'git://github.com/ghoneycutt/puppet-module-common.git'
-      ref: 'v1.0.2'
+      ref: 'v1.3.0'
```
Comment (Phil-Friderici) [inline .fixtures.yml]: common v1.3.0 is needed for Puppet v4 compatibilty

File: spec/classes/init_spec.rb [hunk 2]
```diff
@@ -1998,12 +2006,15 @@
       let(:params) { { :hiera_merge => ['not_a_boolean','or_a_string'] } }
       let(:facts) do
         { :osfamily          => 'RedHat',
+          :fqdn              => 'hieramerge.example.com',
```
Comment (Phil-Friderici) [inline spec/classes/init_spec.rb]: fqdn set to really test merging of hiera data :)

File: spec/classes/init_spec.rb [hunk 3]
```diff
@@ -2032,6 +2046,12 @@
         it { should compile.with_all_deps }
 
         it { should contain_class('ssh') }
+
+        it { should contain_file('sshd_config').with_content(/^\s*DenyUsers denyuser_from_fqdn denyuser_from_common/) }
+        it { should contain_file('sshd_config').with_content(/^\s*DenyGroups denygroup_from_fqdn denygroup_from_common/) }
+        it { should contain_file('sshd_config').with_content(/^\s*AllowUsers allowuser_from_fqdn allowuser_from_common/) }
+        it { should contain_file('sshd_config').with_content(/^\s*AllowGroups allowgroup_from_fqdn allowgroup_from_common/) }
```
Comment (Phil-Friderici) [inline spec/classes/init_spec.rb]: here we test that hiera values get's merged accross multiple levels

File: spec/classes/init_spec.rb [hunk 4]
```diff
@@ -2882,7 +2907,7 @@
       it do
         expect {
           should contain_class('ssh')
-        }.to raise_error(Puppet::Error,/^ssh::ssh_config_global_known_hosts_mode must be a valid 4 digit mode in octal notation. Detected value is <invalidtype>./)
+        }.to raise_error(Puppet::Error,/ssh::ssh_config_global_known_hosts_mode must be a valid 4 digit mode in octal notation\. Detected value is <[\[]?invalid.*type[\]]?/)
```
Comment (Phil-Friderici) [inline spec/classes/init_spec.rb]: with this regex we can test output variants from Puppet v3 & v4

File: spec/fixtures/hiera/hieradata/fqdn/monkey.example.com.yaml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+ssh::sshd_config_allowgroups:
+ssh::sshd_config_allowusers:
+ssh::sshd_config_denygroups:
+ssh::sshd_config_denyusers:
```
Comment (Phil-Friderici) [inline spec/fixtures/hiera/hieradata/fqdn/monkey.example.com.yaml]: seting these to <nil> is needed to pass "with default params" tests

File: spec/spec_helper.rb [hunk 6]
```diff
@@ -1,6 +1,7 @@
 require 'puppetlabs_spec_helper/module_spec_helper'
 
 RSpec.configure do |config|
+  config.hiera_config = 'spec/fixtures/hiera/hiera.yaml'
```
Comment (Phil-Friderici) [inline spec/spec_helper.rb]: this is needed to make use of hiera in rspec

File: .gitignore [hunk 7]
```diff
@@ -26,6 +26,6 @@ doc/
 
 # Puppet
 coverage/
+spec/fixtures/manifests/*
 spec/fixtures/modules/*
 Gemfile.lock
-spec/fixtures/
```
Comment (Phil-Friderici) [inline .gitignore]: need to remove spec/fixtures to make sure hiera files get included

**Comment by ghoneycutt** [comment]
Wow phenomenal work! Thanks a ton, I wouldn't have gotten this without you. Rebased and getting merged in PR #120

**Pull request merged**